packages/components/forms/src/Multiselect/Multiselect.tsx

```diff
@@ -0,0 +1,286 @@
+import React, { useRef, useState } from 'react';
+import { cx } from 'emotion';
+
+import { mergeRefs, type CommonProps } from '@contentful/f36-core';
+import { Button, IconButton } from '@contentful/f36-button';
+import { TextInput, type TextInputProps } from '../TextInput';
+import { CloseIcon, ChevronDownIcon, SearchIcon } from '@contentful/f36-icons';
+import { SkeletonContainer, SkeletonBodyText } from '@contentful/f36-skeleton';
+import { Popover } from '@contentful/f36-popover';
+import { Subheading } from '@contentful/f36-typography';
+
+import { getMultiselectStyles } from './Multiselect.styles';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export interface MultiselectProps
+  extends CommonProps,
+    Pick<
+      TextInputProps,
+      | 'isDisabled'
+      | 'isInvalid'
+      | 'isReadOnly'
+      | 'isRequired'
+      | 'id'
+      | 'defaultValue'
+    > {
+  /** Select Options */
+  children?: React.ReactNode;
+
+  /**
+   * Set a custom icon for the text input
+   */
+  startIcon?: React.ReactElement;
+
+  /**
+   * Placeholder shown before selecting any elements. Defaults to 'Select one or more items'
+   */
+  placeholder?: string;
+
+  /**
+   * current Selected items, to be shown on the trigger button
+   */
+  currentSelection?: Array<string>;
+
+  /**
+   * If this is set to `true` there will be a search input inside the drawer
+   * @default false
+   */
+  hasSearch?: boolean;
+
+  /**
+   * Function called whenever the search input value changes
+   */
+  onSearchValueChange?: (event: React.ChangeEvent) => void;
+
+  /**
+   * This is the value will be passed to the `placeholder` prop of the input.
+   * @default "Search"
+   */
+  searchPlaceholder?: string;
+
+  /**
+   * A message that will be shown when it is not possible to find any option that matches the input value
+   * @default "No matches"
+   */
+  noMatchesMessage?: string;
+
+  /**
+   * Use this prop to get a ref to the input element of the component
+   */
+  searchInputRef?: React.Ref<HTMLInputElement>;
+  /**
+   * Use this prop to get a ref to the toggle button of the component
+   */
+  toggleRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * Use this prop to get a ref to the list of items of the component
+   */
+  listRef?: React.Ref<HTMLUListElement>;
```

Comment:
I'd like to challenge this, it's a reversal to the pattern we moved away from with the breaking changes in v3 to v4. I believe our code conventions show a clear usage of the `subComponentProps={{ ref }}` pattern.

Reply:
I copied these from the autocomplete property list, but i m happy to use the other reference convention.

Reply:
Right, I see. I wish I caught them for the autocomplete then 😅

Reply:
i'll put that on my list of breaking change updates for the autocomplete. 📝